integration/spark/src/main/java/org/apache/carbondata/spark/vectorreader/VectorizedCarbonRecordReader.java

```diff
@@ -348,6 +351,24 @@ private boolean nextBatch() {
     if (iterator.hasNext()) {
       iterator.processNextBatch(carbonColumnarBatch);
       int actualSize = carbonColumnarBatch.getActualSize();
+      if (Boolean.parseBoolean(CarbonProperties.getInstance()
+          .getProperty(CarbonCommonConstants.CARBON_SPARK_VERSION_SPARK3,
+              CarbonCommonConstants.CARBON_SPARK_VERSION_SPARK3_DEFAULT))
+          && null != carbonColumnarBatch.getCarbonWrittenVersion() &&
+          carbonColumnarBatch.getCarbonWrittenVersion().split(CarbonCommonConstants.HYPHEN)[0]
+              .compareTo(CarbonCommonConstants.CARBON_SPARK3_VERSION) < 0) {
+        // rebase Timestamp value
+        CarbonColumnVector[] columnVectors = carbonColumnarBatch.columnVectors;
+        for (CarbonColumnVector vector : columnVectors) {
+          if (DataTypes.TIMESTAMP == vector.getBlockDataType()) {
+            for (int i = 0; i < actualSize; i++) {
+              long timeStamp = vector.getLong(i);
+              long rebaseTime = CarbonToSparkAdapter.rebaseTime(timeStamp);
```

Comment:
here we are getting the filled vectors and then refilling all the values with rebased time again, it may impact the performance for the old stores which will come for rebase, please try to handle such that while filling first time itself, we will fill the rebased timestamp value to save the time and not impact for old stores